cmd/pipecd/Dockerfile

```diff
@@ -1,20 +1,14 @@
 # syntax=docker/dockerfile:1
 # web builder
-# because of this issue, we choose node 18
-# https://github.com/pipe-cd/pipecd/issues/5422
-# https://github.com/nodejs/docker-node/issues/1335#issuecomment-2024344411
-FROM node:18.20.5-alpine3.21 AS web
+# because this stage builds only web assets, we can use any platform
+FROM --platform=$BUILDPLATFORM node:18.20.5-alpine3.21 AS web
```

Comment:
With this modification, we can now use node 20 or later. I want to discuss whether we should upgrade this to current LTS versions.